### ISSUE TITLE
Run on iOS device breaks the application

### ISSUE THREAD

**Issue by vchimev** [NativeScript/nativescript-cli#2723]
The `tns run ios` command breaks the application on an iOS real device. The app crashes with `JS ERROR Error: A Frame must be used to navigate to a Page.`. The same command works fine on iOS simulator.

It is curious that `tns run ios --sync-all-files` works fine on the iOS device.

Steps to reproduce:
```shell
npm install -g nativescript@next # 3.0.0-2017-04-13-8667
tns create appTest && cd appTest
tns run ios # on a connected real device
```

Output:
```
** EXPORT SUCCEEDED **

Project successfully built.
Installing...
Successfully installed on device with identifier 'd97ff43f5d13677a2fb09fc9b4da3066464ff5f4'.
Transferring project files...
Skipping node_modules folder! Use the syncAllFiles option to sync files from this folder.
Successfully transferred all files.
Refreshing application...
Successfully synced application org.nativescript.appTest on device d97ff43f5d13677a2fb09fc9b4da3066464ff5f4.

JS ERROR Error: A Frame must be used to navigate to a Page.
```

Workaround:
```shell
tns run ios --sync-all-files # a connected real device
```

**Comment by vchimev** [comment]
This issue is reproducible with tns-ios: 3.0.0-rc.1-2017-3-28-2.
It has been already fixed in tns-ios@next with https://github.com/NativeScript/ios-runtime/pull/738, and it is not reproducible with tns-ios: 3.0.0-2017-4-18-2.

**Comment by toddanglin** [comment]
To add for others, when using the TypeScript template, the app will hang on load and _not_ display the "A Frame must be used to navigate to a Page" error...but the solution above will work for TypeScript, too.

@vchimev Any chance we'll publish an update to the RC build (rc.2)? Would be good to make some of these critical fixes part of the main RC so more people can test.

**Comment by vchimev** [comment]
@toddanglin, it's done - just published **tns-ios@3.0.0-rc.2** to npm behind the `rc` tag.
